### PR TITLE
Added main ckains and basic API

### PR DESCRIPTION
**Pull request by 71walceli**
- Removed previous approach code

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules
+.*
+.pnpm-store
+Dodkerfile
+**.Dodkerfile
```
Comment (hitchho) [inline .dockerignore]: dodkerfiles probably fine

File: .gitignore [hunk 2]
```diff
@@ -1,25 +1,2 @@
...
-*.log
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
-pnpm-debug.log*
-lerna-debug.log*
-
+.pnpm-store
 node_modules
-dist
```
Comment (hitchho) [inline .gitignore]: we probably dont want to push dist in

File: src/api/config.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+export const config = {
+  chains: {
+    // "polkadot" here can be any unique string value
+    people_rococo_ws: {
+      descriptor: people_rococo,
+      provider: () => getWsProvider("ws://localhost:42001"),
+    },
+    people_rococo: {
+      descriptor: people_rococo,
+      provider: getSmProvider(smoldot.addChain({ chainSpec: JSON.stringify(peopleRococoChainSpec) })),
```
Comment (hitchho) [inline src/api/config.ts]: irrelevant for now. if we want to do this its should not be named people_rococo but just custom_lightclient and chainspec importable from frontend.

File: src/api/config.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    type: "module",
+  }),
+);
+
+export const config = {
+  chains: {
+    // "polkadot" here can be any unique string value
+    people_rococo_ws: {
+      descriptor: people_rococo,
+      provider: () => getWsProvider("ws://localhost:42001"),
```
Comment (hitchho) [inline src/api/config.ts]: here also hardcoding, use should be able to just set any rpc endpoint from frontend  input.
ws:// or wss://

File: src/api/config.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      descriptor: kusama,
+      provider: getSmProvider(smoldot.addChain({ chainSpec: kusamaChainSpec })),
+    },
+    westend: {
+      descriptor: westend,
+      provider: getSmProvider(smoldot.addChain({ chainSpec: westendChainSpec })),
+    },
+    paseo: {
+      descriptor: paseo,
+      provider: getSmProvider(smoldot.addChain({ chainSpec: paseoChainSpec })),
```
Comment (hitchho) [inline src/api/config.ts]: we dont really need to connect any relaychain ever in frontend. users should be interacting with only people parachain for registration.

so polkadot-people, kusama-people, westend-people, paseo-people. if these wont exist as lightlcient, forget the light client and we can use RPC 
people-<relaychain_name>.dotters.network

File: src/api/config.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import { people_rococo, polkadot, kusama, westend, paseo } from "@polkadot-api/descriptors";
+import { WsProvider } from "@polkadot/api";
+import type { Config } from "@reactive-dot/core";
+import { InjectedWalletAggregator } from "@reactive-dot/core/wallets.js";
+//import { chainSpec } from "polkadot-api/chains/polkadot";
+import { chainSpec as polkadotChainSpec } from "polkadot-api/chains/polkadot";
+import { chainSpec as kusamaChainSpec } from "polkadot-api/chains/ksmcc3";
+import { chainSpec as westendChainSpec } from "polkadot-api/chains/westend2";
+import { chainSpec as paseoChainSpec } from "polkadot-api/chains/paseo";
+import peopleRococoChainSpec from "./chainSpecs/bob.json";
```
Comment (hitchho) [inline src/api/config.ts]: naming here bob.json dont really make any sense for json file
yeah i get it its bobs chainspec but i dont get why we have 3 chainspecs name alice bob and people-rococo-local(correct naming and only one probably needed)

**Comment by hitchho** [comment]
```
i❯ user@pc01 𓅨 ~/rotko/w3registrar-www ➤ b660f25|selfMatrixVerify✓
10342 ± yarn                                                                                                                                          [21h30m]✓
! The local project doesn't define a 'packageManager' field. Corepack will now add one referencing yarn@4.5.0+sha512.837566d24eec14ec0f5f1411adb544e892b3454255e61fdef8fd05f3429480102806bac7446bc9daff3896b01ae4b62d00096c7e989f1596f2af10b927532f39.
! For more details about this field, consult the documentation at https://nodejs.org/api/packages.html#packagemanager

➤ YN0000: · Yarn 4.5.0
➤ YN0000: ┌ Resolution step
➤ YN0035: │ @polkadot-api/ink-contracts@npm:0.0.1: Package not found
➤ YN0035: │   Response Code: 404 (Not Found)
➤ YN0035: │   Request Method: GET
➤ YN0035: │   Request URL: https://registry.yarnpkg.com/@polkadot-api%2fink-contracts
➤ YN0000: └ Completed in 1s 698ms
➤ YN0000: · Failed with errors in 1s 710ms
i❯ user@pc01 𓅨 ~/rotko/w3registrar-www ➤ b660f25|selfMatrixVerify⚡
10343 ± bun install                                                                                                                              💥 [21h31m]✹⚡
bun install v1.1.12 (43f0913c)
error: package "web" is not linked

To install a linked package:
   bun link my-pkg-name-from-package-json

Tip: the package name is from package.json, which can differ from the folder name.

i❯ user@pc01 𓅨 ~/rotko/w3registrar-www ➤ b660f25|selfMatrixVerify⚡
10344 ± npm install                                                                                                                              💥 [21h31m]✹⚡
npm error code EUNSUPPORTEDPROTOCOL
npm error Unsupported URL Type "link:": link:polkadot-api/ws-provider/web
npm error A complete log of this run can be found in: /home/user/.npm/_logs/2024-10-12T09_21_04_840Z-debug-0.log
i❯ user@pc01 𓅨 ~/rotko/w3registrar-www ➤ b660f25|selfMatrixVerify⚡
10345 ±      
```

would appreciate if submits in buildable state. here the docker could be useful for reproducible builds

File: src/api/config.ts [hunk 7]
```diff
@@ -12,19 +11,31 @@ import peopleRococoChainSpec from "./chainSpecs/bob.json";
...
+]
 
 export const config = {
   chains: {
-    // "polkadot" here can be any unique string value
-    people_rococo_ws: {
+    custom: {
       descriptor: people_rococo,
-      provider: () => getWsProvider("ws://localhost:42001"),
+      provider: () => getWsProvider(localStorage.wsUrl || "ws://localhost:46085"),
```
Comment (hitchho) [inline src/api/config.ts]: popcli randomizes ports 

ws://127.0.0.1:9944
is default polkadot rpc port

**Comment by 71walceli** [comment]
Oh my... I'll check for that next time. I'll add corresponding Dockerfiles as well...

**Comment by hitchho** [comment]
@71walceli  does not render now since its not connecting correctly to people chains. make it render connecting to people chains intead of relaychains. we dont really need to interact with relaychains at all from frontend. im not sure if the relaychain is also required to be there but you can figure it out from docs ok?

**Review by hitchho — APPROVED** [review]

**Pull request merged**